packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -395,6 +400,127 @@ export class RpcContractProvider extends Provider implements ContractProvider, S
     return new TransactionOperation(hash, content, source, forgedBytes, opResponse, context);
   }
 
+  /**
+   *
+   * @description Stake tz from current address to a specific address. Built on top of the existing transaction operation
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param Stake pseudo-operation parameter
+   */
+  async stake(params: StakeParams) {
+    if (params.to === undefined) {
+      params.to = params.source;
+    }
+    if (params.to !== undefined && params.to !== params.source) {
+      throw new InvalidStakingAddressError(params.to);
+    }
+
+    const sourceValidation = validateAddress(params.source ?? '');
+    if (params.source && sourceValidation !== ValidationResult.VALID) {
+      throw new InvalidAddressError(params.source, invalidDetail(sourceValidation));
+    }
+
+    if (params.amount < 0) {
+      throw new InvalidAmountError(params.amount.toString());
+    }
+    const publicKeyHash = await this.signer.publicKeyHash();
+    const estimate = await this.estimate(params, this.estimator.stake.bind(this.estimator));
+
+    const source = params.source || publicKeyHash;
+    const prepared = await this.prepare.stake({ ...params, ...estimate });
+    const content = prepared.opOb.contents.find(
+      (op) => op.kind === OpKind.TRANSACTION
+    ) as OperationContentsTransaction;
+
+    const opBytes = await this.forge(prepared);
+    const { hash, context, forgedBytes, opResponse } = await this.signAndInject(opBytes);
+    return new TransactionOperation(hash, content, source, forgedBytes, opResponse, context);
+  }
+
+  /**
+   *
+   * @description Unstake tz from current address to a specific address. Built on top of the existing transaction operation
```

Comment:
octez-client desc ref `Unstake the given amount. If "everything" is given as amount, unstakes everything from the staking balance. Unstaked tez remains frozen for a set amount of cycles (the slashing period) after the operation. Once this period is over, the operation "finalize unstake" must be called for the funds to appear in the liquid balance.`